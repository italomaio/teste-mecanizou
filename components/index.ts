export { default as Card } from "./Card/Card";
export { default as RootProviders } from "./RootProviders/RootProviders";
export { default as Logo, type LogoProps } from "./Logo/Logo";
export { default as Input } from "./Input/Input";
export { default as Label, type LabelProps } from "./Label/Label";
export {
  default as Button,
  type ButtonVariant,
  type ButtonSize,
  type ButtonProps,
} from "./Button/Button";
export { default as ThemeToggle } from "./ThemeToggle/ThemeToggle";
export { default as ProductLink } from "./ProductLink/ProductLink";
export { default as Navbar } from "./Navbar/Navbar";
export { default as Skeleton } from "./Skeleton/Skeleton";
export { default as Pagination } from "./Pagination/Pagination";
export { default as ProductRating } from "./ProductRating/ProductRating";
export { default as ProductCard } from "./ProductCard/ProductCard";
export { default as ProductQuantity } from "./ProductQuantity/ProductQuantity";
export { default as ProductImages } from "./ProductImages/ProductImages";
export { default as ProductReviews } from "./ProductReviews/ProductReviews";
export { default as Stars } from "./ProductRating/Stars";
export { default as Loader } from "./Loader/Loader";
export { default as ProductPrices } from "./ProductPrices/ProductPrices";
export {
  default as UserDropdown,
  type UserDropdownProps,
} from "./UserDropdown/UserDropdown";
