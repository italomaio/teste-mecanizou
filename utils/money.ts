export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
