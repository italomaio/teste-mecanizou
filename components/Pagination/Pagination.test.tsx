import { fireEvent, render, screen } from "@/tests/utils";
import { Pagination } from "@/components";

describe("Pagination", () => {
  const onChangePageMock = jest.fn();

  it("Should render correctly", () => {
    const { container } = render(
      <Pagination
        currentPage={1}
        totalPages={3}
        onPageChange={onChangePageMock}
      />
    );

    expect(container).toBeInTheDocument();
  });

  it("Should render 3 pages buttons", () => {
    render(
      <Pagination
        currentPage={1}
        totalPages={3}
        onPageChange={onChangePageMock}
      />
    );

    expect(screen.getAllByTestId(/page/i)).toHaveLength(3);
  });

  it("Should call disable previous button when first page", () => {
    render(
      <Pagination
        currentPage={1}
        totalPages={3}
        onPageChange={onChangePageMock}
      />
    );

    expect(screen.getByLabelText(/previous/i)).toBeDisabled();
  });

  it("Should call disable next button when last page", () => {
    render(
      <Pagination
        currentPage={3}
        totalPages={3}
        onPageChange={onChangePageMock}
      />
    );

    expect(screen.getByLabelText(/next/i)).toBeDisabled();
  });

  it("Should call onPageChange when click on item", () => {
    render(
      <Pagination
        currentPage={3}
        totalPages={3}
        onPageChange={onChangePageMock}
      />
    );

    fireEvent.click(screen.getByRole("button", { name: "Go to page 1" }));
    expect(onChangePageMock).toHaveBeenCalledWith(1);
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(
      <Pagination
        currentPage={3}
        totalPages={3}
        onPageChange={onChangePageMock}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
