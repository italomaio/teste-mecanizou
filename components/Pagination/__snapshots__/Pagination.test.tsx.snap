// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pagination Should match snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Pagination Navigation"
    class="pagination flex flex-row items-center gap-1"
  >
    <button
      aria-label="Go to previous page"
      class="rounded focus-visible-ring inline-flex items-center justify-center gap-2 transition-colors duration-300 cursor-pointer font-medium disabled:cursor-not-allowed hover:bg-gray-100 text-zinc-600 dark:hover:bg-zinc-800 dark:hover:text-white p-2 text-base aspect-square min-h-8"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-left"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m15 18-6-6 6-6"
        />
      </svg>
    </button>
    <button
      aria-label="Go to page 1"
      class="rounded focus-visible-ring inline-flex items-center justify-center gap-2 transition-colors duration-300 cursor-pointer font-medium disabled:cursor-not-allowed hover:bg-gray-100 text-zinc-600 dark:hover:bg-zinc-800 dark:hover:text-white p-2 aspect-square min-h-8 text-xs"
      data-testid="page"
    >
      1
    </button>
    <button
      aria-label="Go to page 2"
      class="rounded focus-visible-ring inline-flex items-center justify-center gap-2 transition-colors duration-300 cursor-pointer font-medium disabled:cursor-not-allowed hover:bg-gray-100 text-zinc-600 dark:hover:bg-zinc-800 dark:hover:text-white p-2 aspect-square min-h-8 text-xs"
      data-testid="page"
    >
      2
    </button>
    <button
      aria-current="page"
      aria-label="Go to page 3"
      class="rounded focus-visible-ring inline-flex items-center justify-center gap-2 transition-colors duration-300 cursor-pointer font-medium disabled:cursor-not-allowed border border-gray-300 bg-gray-50 hover:bg-gray-100 hover:text-accent-foreground dark:border-zinc-700 dark:bg-zinc-800 dark:text-white dark:hover:bg-zinc-700 dark:hover:text-white p-2 aspect-square min-h-8 text-xs"
      data-testid="page"
    >
      3
    </button>
    <button
      aria-label="Go to next page"
      class="rounded focus-visible-ring inline-flex items-center justify-center gap-2 transition-colors duration-300 cursor-pointer font-medium disabled:cursor-not-allowed hover:bg-gray-100 text-zinc-600 dark:hover:bg-zinc-800 dark:hover:text-white p-2 text-base aspect-square min-h-8"
      disabled=""
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </button>
  </nav>
</DocumentFragment>
`;
